src/utils/vfs.js

```diff
@@ -182,7 +191,10 @@ export const humanFileSize = (bytes, si = false) => {
  * @param {string} [options.sortDir='asc'] Sort in this direction
  * @return {Object[]}
  */
-export const transformReaddir = ({path}, files, options = {}) => {
+export const transformReaddir = ({path}, files, capabilityCache, options = {}) => {
```

Comment:
Instead of passing capabilities here, maybe an option could be added here instead ?

Like `options.serverSorting`, which is just passed from `capabilityCache`.

This way we don't introduce any state checks to utility functions.

Reply:
I have committed some changes, although it seems there are still some cognitive complexity.

Reply:
> there are still some cognitive complexity

Feel free to make suggestions on this :)

Reply:
Sorry, but I did not come up with a better solution.